controllers/user.js

```diff
@@ -26,13 +26,33 @@ function removeEmail(req, res) {
   res.redirect("/user/preferences");
 }
 
-function resendEmail(req, res) {
-  // console.log(req.body.emailId);
+async function resendEmail(req, res) {
+  const emailId = req.body.emailId;
+  const sessionUser = _requireSessionUser(req);
+  const existingEmail = await DB.getEmailById(emailId);
+
+  if (existingEmail.subscriber_id !== sessionUser.id) {
+    // TODO: more specific error message?
+    throw new FluentError("user-verify-token-error");
+  }
+
+  const unverifiedEmailAddressRecord = await DB.resetUnverifiedEmailAddress(emailId);
+
+  const email = unverifiedEmailAddressRecord.email;
+  await EmailUtils.sendEmail(
+    email,
+    req.fluentFormat("user-add-email-verify-subject"),
+    "default_email",
+    { email,
+      supportedLocales: req.supportedLocales,
+      verificationHref: EmailUtils.getVerificationUrl(unverifiedEmailAddressRecord),
+      unsubscribeUrl: EmailUtils.getUnsubscribeUrl(unverifiedEmailAddressRecord, "account-verification-email"),
+      whichView: "email_partials/email_verify",
+    }
+  );
 
-  // Resend verification email
-  // Delete the unverified email if it appears anywhere else (under the user's id);
-  // Reset the verification clock
-  return res.json("Resend the email");
+  // TODO: what should we return to the client?
+  return res.json("Resent the email");
```

Comment:
Assuming this is user facing, should it be localized?

Reply:
This isn't user-facing yet. I'm not sure if the status response is in the UI. Whatever is user-facing will be localized though.